### PR TITLE
Migrate `server` module tests to JUnit 5, part 7

### PR DESCRIPTION
**Pull request by dmitrykuzmin**
This PR continues migration of the `server` module to the new JUnit 5 test structure.

The following packages were migrated:
1. `stand`
2. `storage`

### CODE REVIEW

**Comment by dmitrykuzmin** [comment]
@armiol PTAL.

File: core/src/test/java/io/spine/change/LongMismatchTest.java [hunk 1]
```diff
@@ -38,7 +38,7 @@
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
-@SuppressWarnings({"InnerClassMayBeStatic" /* JUnit 5 Nested classes cannot be static */,
+@SuppressWarnings({"InnerClassMayBeStatic" /* JUnit Nested classes cannot be static */,
```
Comment (armiol) [inline core/src/test/java/io/spine/change/LongMismatchTest.java]: Why are they `Nested` while they are `nested`? :)

File: server/src/test/java/io/spine/server/aggregate/AggregateStorageTest.java [hunk 2]
```diff
@@ -145,7 +145,7 @@ protected ProjectId newId() {
             "RefusedBequest",
             "MethodDoesntCallSuperMethod"
     })
-    public void rewrite_record_if_write_by_the_same_id() {
+    protected void rewriteRecord() {
```
Comment (armiol) [inline server/src/test/java/io/spine/server/aggregate/AggregateStorageTest.java]: If this method is a part of the test API (e.g. overridden by the descendants) then please document its purpose.

File: server/src/test/java/io/spine/server/stand/AggregateStateIdStringifierTest.java [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+        final String stringAggregateId = stringifier.convert(id);
+        assertNotNull(stringAggregateId);
+
+        final AggregateStateId restored = stringifier.reverse()
+                                                     .convert(stringAggregateId);
+        assertEquals(id, restored);
+    }
+
+    private static AggregateStateId newStringId() {
```
Comment (armiol) [inline server/src/test/java/io/spine/server/stand/AggregateStateIdStringifierTest.java]: I'd move the helper-factory methods to the `TestEnv`.

File: server/src/test/java/io/spine/server/stand/StandTest.java [hunk 4]
```diff
@@ -1088,8 +1172,8 @@ private void checkEmptyResultForTargetOnEmptyStorage(Query readCustomersQuery) {
         stand.execute(readCustomersQuery, responseObserver);
 
         final List<Any> messageList = checkAndGetMessageList(responseObserver);
-        assertTrue("Query returned a non-empty response message list though the target was empty",
-                   messageList.isEmpty());
+        assertTrue(messageList.isEmpty(),
+                   "Query returned a non-empty response message list though the target was empty");
```
Comment (armiol) [inline server/src/test/java/io/spine/server/stand/StandTest.java]: `had been empty`, I think.

File: server/src/test/java/io/spine/server/stand/given/package-info.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * Test environment classes for the {@code io.spine.server.stand} package.
+ */
+
+@ParametersAreNonnullByDefault
```
Comment (armiol) [inline server/src/test/java/io/spine/server/stand/given/package-info.java]: In `package-info`s we also inclue `@CheckReturnValue` (the one from `com.google.errorprone.annotations`). That helps discovering the API design mistakes.

File: server/src/test/java/io/spine/server/storage/OperatorEvaluatorTest.java [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+        assertTrue(eval(left, EQUAL, right), "basic");
+        assertTrue(eval(right, EQUAL, left), "symmetric");
+        assertTrue(eval(left, EQUAL, left), "reflective");
+        assertTrue(eval(left, EQUAL, third) || eval(third, EQUAL, right), "transitive");
+        assertTrue(eval(null, EQUAL, null), "nullable");
+        assertTrue(eval(left, EQUAL, right), "consistent");
+    }
+
+    @Test
+    @DisplayName("compare not equal instances")
```
Comment (armiol) [inline server/src/test/java/io/spine/server/storage/OperatorEvaluatorTest.java]: `non-equal`

File: server/src/test/java/io/spine/server/storage/given/RecordStorageTestEnv.java [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+        public void delete() {
+            TestTransaction.delete(this);
+        }
+    }
+
+    /**
+     * The {@link TestCounterEntity} {@linkplain Project.Status project status} represented by the
+     * {@linkplain Enum Java Enum}.
+     *
+     * <p>Needed to check filtering by the {@link Enumerated} entity columns.
```
Comment (armiol) [inline server/src/test/java/io/spine/server/storage/given/RecordStorageTestEnv.java]: `It is used to test the entity filtering when the query is built around `{@link Enumerated} entity columns.`

However, I am not sure why we don't use the Java enum generated from the Protobuf definition, but instead we duplicate the definition.

**Pull request merged**